modules/nf-core/ashlar/main.nf

```diff
@@ -0,0 +1,53 @@
+process ASHLAR {
+    tag '$meta.id'
+    label 'process_single'
+
+    conda "bioconda::ashlar=1.17.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/ashlar:1.17.0--pyh5e36f6f_0' :
+        'biocontainers/ashlar:1.17.0--pyh5e36f6f_0' }"
+
+    input:
+    tuple val(meta), path(images)
+    path(opt_dfp)
+    path(opt_ffp)
+
+    output:
+    tuple val(meta), path("*.ome.tif"), emit: tif
+    path "versions.yml"               , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def dfp = opt_dfp ? "--dfp $opt_dfp" : ""
+    def ffp = opt_ffp ? "--ffp $opt_ffp" : ""
+    def num_files = images instanceof List ? images.size() : 1
+    def opt_dfp_size = opt_dfp instanceof List ? opt_dfp.size() : 1
+    def opt_ffp_size = opt_ffp instanceof List ? opt_ffp.size() : 1
+    def dfp_validated = (opt_dfp_size == 0 || opt_dfp_size == 1 || opt_dfp_size == num_files) ? true : false
+    def ffp_validated = (opt_ffp_size == 0 || opt_ffp_size == 1 || opt_ffp_size == num_files) ? true : false
+
+    if ( !dfp_validated ) { error "Please input only zero, one, or N dfp files, where N is the number of input images" }
+    if ( !ffp_validated ) { error "Please input only zero, one, or N ffp files, where N is the number of input images" }
+
+    """
+
+    ashlar \\
+        -o ${prefix}.uuid.ome.tif \\
+        $images \\
+        $args \\
+        $dfp \\
+        $ffp
+
+    sed -E 's/UUID="urn:uuid:[[:xdigit:]]{8}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{12}"/UUID="urn:uuid:00000000-0000-0000-0000-000000000000"/g' ${prefix}.uuid.ome.tif > ${prefix}.ome.tif
```

Comment:
I have no idea whether any downstream tool may care about it or not.
I considered just finding and replacing everything after the `urn:uuid`, but didn't want to accidentally match anything else.

Reply:
@SPPearce Can you comment on why you changed from `sed -i` (in-place) to operating on a copy? These images can be hundreds of GBs so we try to avoid making copies where possible.

Reply:
The `-i` flag didn't seem to be working on the CI for some reason. I definitely agree that would be better.

One thought is that the metadata is at the very end of the tif from what I could see, so there might be some clever sed (or similar) magic that could be used to start from the end of a file and only make one change.

Reply:
Oh, the test.yml was broken in your commit that failed CI. The file tests were changed from `md5sum` to `contains` with the placeholder TODO message. That's since been fixed so we're going to try again with `sed -i`.

Reply:
It now passes all tests.
Thank you for figuring that out @jmuhlich!